airflow/models/taskinstance.py

```diff
@@ -2306,12 +2306,14 @@ def emit_state_change_metric(self, new_state: TaskInstanceState) -> None:
         elif new_state == TaskInstanceState.QUEUED:
             metric_name = "scheduled_duration"
             if self.start_date is None:
-                # same comment as above
-                self.log.warning(
-                    "cannot record %s for task %s because previous state change time has not been saved",
-                    metric_name,
-                    self.task_id,
-                )
+                # commented out because this waning is not working correctly
+                # until field like a `scheduled_dttm` are implemented
+                # see #30612 #34493 and #34771 for more details
+                # self.log.warning(
+                #     "cannot record %s for task %s because previous state change time has not been saved",
+                #     metric_name,
+                #     self.task_id,
+                # )
```

Comment:
```suggestion
                # chaning the logging level to debug because this waning is not working correctly
                # until field like a `scheduled_dttm` are implemented
                # see #30612 #34493 and #34771 for more details
                self.log.warning(
                    "cannot record %s for task %s because previous state change time has not been saved",
                    metric_name,
                    self.task_id,
                )
```

Looks like the consensus is to change the logging level to debug. Adding this suggestion for easier update